src/tiled/tileseteditor.cpp

```diff
@@ -465,7 +465,28 @@ void TilesetEditor::addTiles()
         Tile *newTile = new Tile(tileset->takeNextTileId(), tileset);
         newTile->setImage(loadedFile.image);
         newTile->setImageSource(loadedFile.imageSource);
-        tiles.append(newTile);
+
+        // If the tile is already in the tileset, warn user and confirm addition
+        bool addImage = true;
+        const QMap<int, Tile*> addedTiles = mCurrentTilesetDocument->tileset().data()->tiles();
+        for(auto tile : addedTiles) {
+            if(tile->imageSource() == newTile->imageSource()) {
+                QMessageBox warning(QMessageBox::Warning,
+                            tr("Add Tiles"),
+                            tr("Tile already exists in the Tileset!"),
+                            QMessageBox::Yes | QMessageBox::No,
```

Comment:
I added the checkbox with "Apply this action to all tiles", similar to what shows on cinnamon, seemed more appropriate than "Don't show this again" from Windows Explorer.

A choice that I have made is that this checkbox is valid only for the images being added, not for all future actions, as the user does not have a place to configure this preferences and would not be able to revert this behaviour.